### PR TITLE
[ECO-676] Modules rearrangement + Some bug fixes and improvements

### PR DESCRIPTION
**Pull request by hernantain**
**What**:

1. Extracting common logic into a base handler class to avoid repeated code. 
2. Moving handlers classes into a handlers package to avoid having files flying around. 
3. Renaming `fetch_one` method on `Query` class to `execute`
4. Fixing dates validation in `SeatrchIndicator` class.
5. `use_empty_paylaod` param in `Query` class to be able to search indicators with an empty payload.
6. Differentiating payload params from query params to simplify logic. (before we had only one attibute "params")
7. Forcing MAX values in some classes. (e.g. 2000 IOCs per submission, 50 attributes per IOC, 20 tags per IOCs, etc)
8. Adding dateparser into package dependencies.

**Why**:

To fix bugs and improve SDK validations, code, usability

**How**: 

Writing code 🤷‍♂️

### CODE REVIEW

**Review by imonaco — APPROVED** [review]
Looking good! First time looking at this code , so I can't detect any critical part of it been changed , but I don't see any code smells either.

**Review by marcos-vivar — APPROVED** [review]
LGTM! Amazing work @hernantain 🚀 
:shipit:

File: trustar2/handlers/safelist.py [hunk 1]
```diff
@@ -1,34 +1,23 @@
...
-from .trustar_enums import ObservableTypes
+from trustar2.base import fluent, Methods, ParamsSerializer, Param, get_timestamp
+from trustar2.handlers.base_handler import BaseHandler
+from trustar2.query import Query
+from trustar2.trustar_enums import ObservableTypes
 
 
 @fluent
-class Safelist(object):
+class Safelist(BaseHandler): # FALTA
```
Comment (manuelramos) [inline trustar2/handlers/safelist.py:10]: What is the meaning of this comment? maybe it's the acronym for something I don't know, that's why I ask.

**Review by manuelramos — APPROVED** [review]
looks good! great job. I just left an small comment but I'm approving this PR anyway.

File: trustar2/handlers/safelist.py (hunk 1 above)
Comment (hernantain) [reply trustar2/handlers/safelist.py:10]: hahah no, thanks for noticing

**Pull request merged**